pkg/controller/storagecluster/reconcile.go

```diff
@@ -409,3 +466,80 @@ func newCephDaemonResources(custom map[string]corev1.ResourceRequirements) map[s
 
 	return resources
 }
+
+// newStorageClassDeviceSets converts a list of StorageDeviceSets into a list of Rook StorageClassDeviceSets
+func newStorageClassDeviceSets(storageDeviceSets []ocsv1.StorageDeviceSet, topologyMap *ocsv1.NodeTopologyMap) []rook.StorageClassDeviceSet {
+	var storageClassDeviceSets []rook.StorageClassDeviceSet
+
+	for _, ds := range storageDeviceSets {
+		resources := ds.Resources
+		if resources.Requests == nil && resources.Limits == nil {
+			resources = defaults.DaemonResources["osd"]
+		}
+
+		topologyKey := ds.TopologyKey
+		topologyKeyValues := []string{}
+		noPlacement := ds.Placement.NodeAffinity == nil && ds.Placement.PodAffinity == nil && ds.Placement.PodAntiAffinity == nil
+
+		if noPlacement {
+			if topologyKey == "" {
+				topologyKey = "zone"
+			}
+			if topologyMap != nil {
+				topologyKey, topologyKeyValues = topologyMap.GetKeyValues(topologyKey)
+			}
+		}
+
+		replica := ds.Replica
+		if replica == 0 {
+			replica = defaults.DeviceSetReplica
+		}
+
+		// This is a temporary hack in place due to limitations in
+		// the current implementation of the OCP console.
+		// TODO: Remove this behavior when OCP console is updated
+		count := ds.Count / 3
+
+		for i := 0; i < replica; i++ {
+			placement := rook.Placement{}
+			portable := ds.Portable
+
+			if noPlacement {
```

Comment:
I think it would be worthwhile to document some of this default Placement behavior in the top level ocs-operator README.md file soon.

Reply:
I'll do it closer to release, once we're sure the placement logic has stabilized.

Reply:
Created Issue #192.

Reply:
excellent, thanks